membership-attribute-service/app/services/ScanamoFeatureToggleService.scala

```diff
@@ -0,0 +1,25 @@
+package services
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBAsyncClient
+import com.gu.scanamo._
+import com.gu.scanamo.error.DynamoReadError
+import com.gu.scanamo.syntax.{set => scanamoSet, _}
+import com.typesafe.scalalogging.LazyLogging
+import models.FeatureToggle
+import play.api.libs.concurrent.Execution.Implicits._
+
+import scala.concurrent.Future
+
+class ScanamoFeatureToggleService(client: AmazonDynamoDBAsyncClient, table: String) extends LazyLogging {
+
+  private val scanamo = Table[FeatureToggle](table)
+  def run[T] = ScanamoAsync.exec[T](client) _
+
+  def get(featureName: String): Future[Option[FeatureToggle]] =
+    run(scanamo.get('FeatureName -> featureName).map(_.flatMap {
+      _
+        .left.map(e => logger.warn(s"Scanamo error in get: ${DynamoReadError.describe(e)}", e))
```

Comment:
I'd rather return the error in a left and let someone upstream log it, as this pattern is the root of a few problems we had in membership-common.  also we separate the error from its context, so we don't know what it was trying to do at the time, we have to find two warnings at a similar time in the logs